src/tests/7_Follow.test.ts

```diff
@@ -0,0 +1,186 @@
+import { expect } from "chai";
+import UserService from "../db/service/user.service";
+import UserInfoService from "../db/service/userInfo.service";
+import FollowService from "../modules/follows/FollowService";
+import UserCollection, { UserDocument } from "../db/models/user.model";
+import UserInfoCollection, { UserInfo } from "../db/models/userInfo.model";
+
+import { cleanupMockedUserInfo, createAndTestUserInfo } from "./2_userInfo.test";
+import { generateRandomPassword } from "../utils/utils";
+
+interface MockedUser {
+	mockedUser: UserDocument;
+	mockedInfo: UserInfo;
+}
+
+describe("Follow service can", () => {
+	let followService: FollowService;
+	let mockedFollower: MockedUser;
+	let mockedUser: MockedUser;
+	let userService: UserService;
+	let userInfoService: UserInfoService;
+
+	describe("setup", () => {
+		it("instantiate the service class", () => {
+			followService = new FollowService();
+			expect(followService).not.equal(undefined);
+		});
+
+		it("create prerequesite services", () => {
+			userService = new UserService(UserCollection);
+			userInfoService = new UserInfoService(UserInfoCollection);
+			expect(userService).not.equal(undefined);
+			expect(userInfoService).not.equal(undefined);
+		});
+
+		it("create two mocked users", async () => {
+			const mockedUserInput = {
+				createUserInput: {
+					email: "mocked@email.com",
+					password: await generateRandomPassword(),
+					username: "mockedUser",
+				},
+				createInfoInput: {
+					firstname: "Mocked",
+					lastname: "User",
+					email: "mocked@email.com",
+					username: "mockedUser",
+				},
+			};
+
+			const mockedFollowerInput = {
+				createUserInput: {
+					email: "mockedFollower@email.com",
+					password: await generateRandomPassword(),
+					username: "mockedFollower",
+				},
+				createInfoInput: {
+					firstname: "Mocked",
+					lastname: "Follower",
+					email: "mockedFollower@email.com",
+					username: "mockedFollower",
+				},
+			};
+
+			mockedFollower = await createAndTestUserInfo({
+				createUserInfoInput: mockedFollowerInput.createInfoInput,
+				createUserInput: mockedFollowerInput.createUserInput,
+				userInfoService: userInfoService,
+				userService: userService,
+			});
+
+			mockedUser = await createAndTestUserInfo({
+				createUserInfoInput: mockedUserInput.createInfoInput,
+				createUserInput: mockedUserInput.createUserInput,
+				userInfoService: userInfoService,
+				userService: userService,
+			});
+		});
+	});
+
+	describe("follow users", () => {
+		it("follow another user", async () => {
+			// mockedFollower sends request to mockedUser
+			const result = await followService.follow(
+				mockedFollower.mockedInfo.userId.toJSON(),
+				mockedUser.mockedInfo.userId.toJSON()
+			);
+			expect(result.success).to.be.true;
```

Comment:
Can you test if the user actually has the mockedFollower as a follower? 

If not it's fine but it would make the test more effective

Reply:
I already made that in the below tests for `getFollows(..)` and `getFollowers(..)`